streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -285,18 +299,13 @@ void handleLostAll() {
         final Iterator<Task> iterator = tasks.values().iterator();
         while (iterator.hasNext()) {
             final Task task = iterator.next();
-            final Set<TopicPartition> inputPartitions = task.inputPartitions();
             // Even though we've apparently dropped out of the group, we can continue safely to maintain our
             // standby tasks while we rejoin.
             if (task.isActive()) {
+                cleanupTask(task);
                 task.closeDirty();
-                changelogReader.remove(task.changelogPartitions());
-            }
-
-            for (final TopicPartition inputPartition : inputPartitions) {
```

Comment:
Not sure why we do not remove the input partitions previously.. is that intentional @vvcephei ? If not I'd move the removal into the block (I already did just to clarify :P)

Reply:
The intent was to remove the input partitions from the map any time we remove a task from `tasks`. It looks like your code maintains this (in a clearer and cleaner way).

Reply:
Got it. The previous code is that we would `partitionToTask.remove(inputPartition);` and remove the task no matter if the task is closed or not, which is a bit weird --- for standby tasks, we do not close them, but we still remove from the iterator and remove from the materialized `partitionToTask`. My modification is to ONLY do this logic if we are closing the task.

As long as you agree this is correct I'm relieved.

Reply:
Ah, then it was my mistake before! Good catch.